polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -1449,6 +1498,30 @@ pub mod pallet {
 			})?;
 			Ok(())
 		}
+
+		#[pallet::call_index(13)]
+		#[pallet::weight(T::WeightInfo::execute_blob())]
+		pub fn execute_blob(
+			origin: OriginFor<T>,
+			message: BoundedVec<u8, MaxXcmEncodedSize>,
+			max_weight: Weight,
+		) -> DispatchResultWithPostInfo {
+			let weight_used = <Self as ExecuteController<_, _>>::execute_blob(
```

Comment:
you don't do this here (and in send_blob) below?
```
			let origin_location = T::ExecuteXcmOrigin::ensure_origin(origin)?;
```

Reply:
That's already done in `ExecuteController::execute_blob` (and `SendController::send_blob`)